recipes/runtimeqml/all/conanfile.py

```diff
@@ -0,0 +1,70 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake
+from conan.tools.files import get, copy
+from conan.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.49.0"
+
+class RuntimeQml(ConanFile):
+    name = "runtimeqml"
+    homepage = "https://github.com/GIPdA/runtimeqml"
+    description = "Enables hot-reloading qml files"
+    topics = ("qt", "hot-reload", "qml", "gui")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSD-3-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+
+    generators = "CMakeDeps", "CMakeToolchain"
+
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][str(self.version)],
+            destination=self.source_folder, strip_root=True)
+
+    def requirements(self):
+        if self.version == "cci.20211220": # Only version which supports qt5
+            self.requires("qt/5.15.5")
+        else:
+            self.requires("qt/6.3.1")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        qt = self.dependencies["qt"]
+        if not qt.options.qtdeclarative:
+            raise ConanInvalidConfiguration(f"{self.ref} requires option qt:qtdeclarative=True")
```

Comment:
```suggestion
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            try:
                del self.options.fPIC
            except Exception:
                pass

    def layout(self):
        cmake_layout(self)
            
    def requirements(self):
        if Version(self.version) <= "cci.20211220"
            self.requires("qt/5.15.5")
        else:
            self.requires("qt/6.3.1")

    def validate(self):
        if self.info.settings.compiler.cppstd:
            check_min_cppstd(self, self._minimum_cpp_standard)
        check_min_vs(self, 191)
        if not is_msvc(self):
            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
                raise ConanInvalidConfiguration(
                    f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support."
                )
        qt = self.dependencies["qt"]
        if not qt.options.qtdeclarative:
            raise ConanInvalidConfiguration(f"{self.ref} requires option qt:qtdeclarative=True")

    def source(self):
        get(self, **self.conan_data["sources"][str(self.version)],
            destination=self.source_folder, strip_root=True)
```
Re-ordered methods to follow conan create flow: https://docs.conan.io/en/latest/reference/commands/creator/create.html#methods-execution-order

Add layout method

It requires C++17